scene/resources/tile_set.cpp

```diff
@@ -4835,9 +4844,10 @@ void TileData::add_custom_data_layer(int p_to_pos) {
 
 void TileData::move_custom_data_layer(int p_from_index, int p_to_pos) {
 	ERR_FAIL_INDEX(p_from_index, custom_data.size());
-	ERR_FAIL_INDEX(p_to_pos, custom_data.size() + 1);
-	custom_data.insert(p_to_pos, navigation[p_from_index]);
-	custom_data.remove_at(p_to_pos < p_from_index ? p_from_index + 1 : p_from_index);
+	ERR_FAIL_INDEX(p_to_pos, custom_data.size());
+	Variant data = custom_data[p_from_index];
+	custom_data.remove_at(p_from_index);
+	custom_data.insert(p_to_pos, data);
```

Comment:
this also fixes a sneaky bug. previously, the navigation data was used instead of custom data